django/utils/inspect.py

```diff
@@ -50,12 +50,16 @@ def func_accepts_var_args(func):
     )
 
 
-def func_has_no_args(func):
-    args = [
-        p for p in inspect.signature(func).parameters.values()
+def method_has_no_args(meth):
```

Comment:
Consider this rename and change in a separate commit as it seems not entirely related to the paginator changes. Also, tests for paginator are missing.

Reply:
It is related. The other place this is used is in `admindocs`. The functions it is passed are of the nature `Model.method` while in `Paginator` I am checking against `instance.method`. The way `self` is treated in the signature is different between methods referenced from the instance or the class. The name of the function also makes no sense:

```python
>>> def foo():
>>>    pass

>>> func_has_no_args(foo)
False
```

At least it's obvious that the function is now called on methods. I am guessing it was given that name since in `admindocs` it's within an `isfunction` guard and that returns `False` for bound methods.

Reply:
I meant that the change could stand alone (precede) the paginator changes. If we had to revert the paginator changes, the rename still makes sense.